include/mbedtls/ccm.h

```diff
@@ -40,6 +42,106 @@ typedef struct {
 }
 mbedtls_ccm_context;
 
+#else /* MBEDTLS_CCM_ALT */
+#include "ccm_alt.h"
+
+typedef struct {
+    union
+    {
+        mbedtls_ccm_alt_context_t ccm_alt_ctx;    /*!< opaque context structure to be defined in ccm_alt.h */
+        mbedtls_cipher_context_t cipher_ctx;
+    }u_ctx;
+    unsigned int    is_alternative;
+}
```

Comment:
@hanno-arm You are correct, and I will add a check for `ctx->is_alternative==0` at the `mbedtls_ccm_free` before calling `mbedtls_cipher_free( cipher_ctx );`. Thanks for the comment.
The idea to add this restriction, is to have a SW fallback, in case the HW engine does not support the specific use case, and we want to avoid the need of copying SW implementation code only to support the specific code (like currently we have in AES use case)